src/ifcopenshell-python/ifcopenshell/api/pset/edit_pset.py

```diff
@@ -275,7 +299,12 @@ def add_new_properties(self):
                             EnumerationReference=prop_enum,
                         )
                         properties.append(prop_enum_value)
-                        continue
+                        break
+
+                    raise NotImplementedError(f"Template type '{pset_template.TemplateType}' is not supported yet")
+
+                # TODO: what if no template was found/matched?
```

Comment:
I vote for foreseeing all possible cases (bounded properties, etc.) and throwing a NotImplementedError while they're not implemented.